### PR TITLE
Remove round 0 from AY 24/25 onwards

### LINKED ISSUE #36: Remove round 0 column for year 24/25 onwards
**Describe the problem**
Round 0 has been removed from the system from the year 24/25.
From the front end, we can stop displaying the round 0 column for this academic year onwards as it is redundant information.

**Describe the feature**
Remove the round 0 column for year 24/25 onwards.

### PR DESCRIPTION
**Pull request by leslieyip02**
## Description

Closes #36. Removes round 0 column if the year is not `"2122"`, `"2223"` or `"2324"`.

### AY 24/25

Screenshot 2024-12-01 214919

### AY 23/24

Screenshot 2024-12-01 214926

### CODE REVIEW

File: src/web/templates/history.html [hunk 1]
```diff
@@ -99,6 +99,7 @@ <h1 style="font-size: 26px; font-weight: bold; font-family: 'Roboto', sans-serif
       {% set year = request.form.get('year', '2425') %}
       {% set semester = request.form.get('semester', '1') %}
       {% set type = request.form.get('type', 'ug') %}
+      {% set start_round_num = 0 if year in ['2122', '2223', '2324'] else 1 %}
```
Comment (lunaflight) [inline src/web/templates/history.html:102]: Is it possible to import the constants `PRE_2425_ROUNDS`, `POST_2425_ROUNDS` and `PRE_2425_YEARS` via `context_processor()` in `src/web/app.py`?

Then, we can reuse the logic by saying something like `set round_nums = PRE_2425_ROUNDS if year in PRE_2425_YEARS else POST_2425_ROUNDS` so there is less code duplication. On line 111, the `round_num` can then be directly derived from `round_nums`.

**Review by lunaflight — CHANGES_REQUESTED** [review]
Hey, thanks for the PR! This looks really nice, I just have 1 minor thought about logic duplication in `history.html` - let me know if it's possible to implement this.

File: src/history/api.py [hunk 2]
```diff
@@ -91,6 +95,23 @@ def _clean_code(code: str) -> str:
 CourseData = dict[str, Union[str, ClassDict]]
 
 
+def get_round_numbers(year: Union[str, int]) -> tuple[int]:
```
Comment (lunaflight) [inline src/history/api.py:98]: Small nitpick - upon running `./scripts/checks.sh`, `mypy` complains about the following:
```
src/history/api.py:112: error: Incompatible return value type (got "tuple[int, ...]", expected "tuple[int]")  [return-value]
```
The correct type signature should return a `tuple[int, ...]` and everything else should be fine.

**Review by lunaflight — CHANGES_REQUESTED** [review]
Just 1 final type-checking error to resolve and I'm ready to merge it!

File: src/history/api.py (hunk 2 above)
Comment (leslieyip02) [reply src/history/api.py:98]: Oops, my bad.

**Review by lunaflight — APPROVED** [review]
Looks good to me, squashed, merged and pushed to prod :)

**Pull request merged**